pkg/envoy/ads/response_benchmark_test.go

```diff
@@ -0,0 +1,170 @@
+					Enable: false,
+				},
+			},
+			FeatureFlags: configv1alpha2.FeatureFlags{
+				EnableWASMStats:    false,
+				EnableEgressPolicy: false,
+			},
+		},
+	}
+	_, err = configClient.ConfigV1alpha2().MeshConfigs(tests.OsmNamespace).Create(context.Background(), &meshConfig, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create mesh config: %v", err)
+	}
+
+	certManager, err := certificate.FakeCertManager()
+	if err != nil {
+		b.Fatalf("Failed to create fake cert manager: %v", err)
+	}
+
+	// --- setup
+	namespace := tests.Namespace
+	proxyService := service.MeshService{
+		Name:      tests.BookstoreV1ServiceName,
+		Namespace: namespace,
+	}
+	proxySvcAccount := tests.BookstoreServiceAccount
+
+	certPEM, _ := certManager.IssueCertificate(proxySvcAccount.ToServiceIdentity().String(), certificate.Service)
+	cert, _ := certificate.DecodePEMCertificate(certPEM.GetCertificateChain())
+	server, _ := tests.NewFakeXDSServer(cert, nil, nil)
+
+	proxyUUID := uuid.New()
+	labels := map[string]string{constants.EnvoyUniqueIDLabelName: proxyUUID.String()}
+	meshSpec := smi.NewSMIClient(informerCollection, tests.OsmNamespace, kubeController, msgBroker)
+	mc := catalog.NewMeshCatalog(
+		kubeController,
+		meshSpec,
+		certManager,
+		policyController,
+		stop,
+		osmConfigurator,
+		[]service.Provider{kubeProvider},
+		[]endpoint.Provider{kubeProvider},
+		msgBroker,
+	)
+
+	proxyRegistry := registry.NewProxyRegistry(registry.ExplicitProxyServiceMapper(func(*envoy.Proxy) ([]service.MeshService, error) {
+		return nil, nil
+	}), nil)
+
+	pod := tests.NewPodFixture(namespace, fmt.Sprintf("pod-0-%s", proxyUUID), tests.BookstoreServiceAccountName, tests.PodLabels)
+	pod.Labels[constants.EnvoyUniqueIDLabelName] = proxyUUID.String()
+	_, err = kubeClient.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create pod: %v", err)
+	}
+
+	// monitor namespace
+	nsObj := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   namespace,
+			Labels: map[string]string{constants.OSMKubeResourceMonitorAnnotation: tests.MeshName},
+		},
+	}
+	if err := informerCollection.Add(informers.InformerKeyNamespace, &nsObj, &testing.T{}); err != nil {
+		b.Fatalf("Failed to add namespace to informer collection: %s", err)
+	}
+
+	svc := tests.NewServiceFixture(proxyService.Name, namespace, labels)
+	_, err = kubeClient.CoreV1().Services(namespace).Create(context.Background(), svc, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create service: %v", err)
+	}
+
+	proxy := envoy.NewProxy(envoy.KindSidecar, proxyUUID, proxySvcAccount.ToServiceIdentity(), nil)
+
+	s := NewADSServer(mc, proxyRegistry, true, tests.Namespace, osmConfigurator, certManager, kubeController, nil)
+
+	// Set subscribed resources for SDS
+	proxy.SetSubscribedResources(envoy.TypeSDS, mapset.NewSetWith("service-cert:default/bookstore", "root-cert-for-mtls-inbound:default/bookstore"))
+
```

Comment:
can you offload everything above this line into a `setupTestServer` func? Makes it a bit more clear to see what we're testing